staking/types/validator.go

```diff
@@ -74,6 +74,16 @@ type ValidatorStats struct {
 	BLSKeyPerShard []KeysPerShard
 }
 
+// ValidatorSnapshot ..
+type ValidatorSnapshot struct {
+	Validator *ValidatorWrapper
+	Epoch     *big.Int
+	// The number of blocks the validator should've signed when in active mode (selected in committee)
+	NumBlocksToSign *big.Int `rlp:"nil"`
```

Comment:
Let's just put NumBlocksToSign NumBlocksSigned into validatorWrapper, which is in state. It will be needed for fast state sync.

Reply:
It's better to have a permanent record of a validator's signing records.